### PR TITLE
Extend mobile browser detection

### PR DESCRIPTION
**Pull request by tremby**
On top of the existing detections of Android and IOS, detect Windows Phone, IE Mobile, Blackberry and BB10.

As discussed in https://github.com/harvesthq/chosen/issues/1848 and https://github.com/harvesthq/chosen/issues/2439

### CODE REVIEW

**Comment by tremby** [comment]
One thing I question is whether the tests should really be case-insensitive. I would lean towards them not being. Maybe it doesn't make much difference.

File: coffee/lib/abstract-chosen.coffee [hunk 1]
```diff
@@ -282,12 +282,20 @@ class AbstractChosen
   # class methods and variables ============================================================
 
   @browser_is_supported: ->
-    if window.navigator.appName == "Microsoft Internet Explorer"
-      return document.documentMode >= 8
     if /iP(od|hone)/i.test(window.navigator.userAgent)
       return false
     if /Android/i.test(window.navigator.userAgent)
       return false if /Mobile/i.test(window.navigator.userAgent)
+    if /IEMobile/i.test(navigator.userAgent)
```
Comment (tjschuck) [inline coffee/lib/abstract-chosen.coffee]: Looks like you're missing a `window.` on all of these conditions.

Comment (tremby) [inline coffee/lib/abstract-chosen.coffee]: It would still work, would it not? window being the global object and all. But I'll add it for consistency.

Comment (tremby) [inline coffee/lib/abstract-chosen.coffee]: Replaced the commit.

**Comment by tjschuck** [comment]
This looks good to me, but hopefully we can get a :+1: from a more qualified @harvesthq/chosen-developers -- maybe @koenpunt if we're lucky?

**Comment by koenpunt** [comment]
I'm not aware of the actual user agent strings, but this seems good to me

**Comment by tjschuck** [comment]
Thanks @tremby and @StephKoenig (and @koenpunt)!

**Pull request merged**